encoder/vaapiencoder_vp9.cpp

```diff
@@ -280,10 +312,25 @@ bool VaapiEncoderVP9::referenceListUpdate(const PicturePtr& pic,
     if (pic->m_type == VAAPI_PICTURE_I) {
         m_reference.clear();
         m_reference.insert(m_reference.end(), kMaxReferenceFrames, recon);
+        if (getReferenceMode())
+            m_currentReferenceIndex = 0;
     }
     else {
-        m_reference.pop_front();
-        m_reference.push_front(recon);
+        if (getReferenceMode()) {
+            ReferenceQueue::iterator it
+                = m_reference.begin() + m_currentReferenceIndex;
+
+            (*it) = recon;
+
+#if __ENABLE_DEBUG__
```

Comment:
No, this cannot be removed as we don't want the for cycle to do a no-op. I find printing the reference frame useful in debug mode but if it must be removed then I'll remove the whole thing.